onnxruntime/test/providers/xnnpack/xnnpack_basic_test.cc

```diff
@@ -89,6 +91,51 @@ TEST(XnnpackEP, TestNhwcConvReluClipFusion) {
   RunAndVerifyOutputsWithEP(ort_model_path, "TestNhwcConvReluClipFusion", std::move(ep), feeds, params);
 }
 
+#ifdef XNNPACK_FP16_SUPPORTED
+TEST(XnnpackEP, TestNhwcConvReluClipFusion_FP16) {
+  const ORTCHAR_T* ort_model_path = ORT_MODEL_FOLDER "nhwc_conv_clip_relu_fp16.onnx";
+
+  RandomValueGenerator generator;
+  TensorShape input_shape_x{1, 16, 16, 192};
+  std::vector<MLFloat16> input_x = generator.Uniform<MLFloat16>(input_shape_x.GetDims(), -128, 128);
+
+  OrtValue ml_value_x;
+  CreateMLValue<MLFloat16>(input_shape_x.GetDims(), input_x.data(), OrtMemoryInfo(), &ml_value_x);
+
+  NameMLValMap feeds;
+  feeds.insert(std::make_pair("model_input", ml_value_x));
+
+  std::function<void(const Graph&)> verify = [](const Graph& graph) -> void {
+    ASSERT_EQ(graph.NumberOfNodes(), 3) << "Transpose nodes should have been removed, and "
+                                           "Conv+Relu and Conv+Clip should have been fused, leaving 3 nodes.";
+    auto node_iter = graph.Nodes().begin();
+    auto check_node = [](const Node& node, const std::string& fusion_type) {
+      const auto& attr = node.GetAttributes();
+      auto activation = attr.find("activation");
+      ASSERT_NE(activation, attr.cend()) << "Fused node should have activation attribute";
+      ASSERT_EQ(activation->second.s(), fusion_type);
+    };
+
+    // check 2nd and 3rd nodes.
+    // the first node is the Conv that does not get fused (created after first call to GetCapability)
+    // the 2nd and 3rd nodes are the fused nodes (created after second call to GetCapability)
+    ++node_iter;
+    check_node(*node_iter, "Clip");
+    ++node_iter;
+    check_node(*node_iter, "Relu");
+  };
+
+  EPVerificationParams params;
+  params.ep_node_assignment = ExpectedEPNodeAssignment::All;
+  params.fp32_abs_err = 0.0002f;
+  params.graph_verifier = &verify;
+
+  auto ep = DefaultXnnpackExecutionProvider();
+  // So far, CPU EP doensn't support Fp16 Conv fusion, so verify_outputs is skipped.
+  RunAndVerifyOutputsWithEP(ort_model_path, "TestNhwcConvReluClipFusion_FP16", std::move(ep), feeds, params, {}, false);
```

Comment:
Not quite following. There should still be valid output from the CPU EP even if it doesn't fuse, so why can't we use verify_outputs? 

```suggestion
  // So far, CPU EP doesn't support Fp16 Conv fusion, so verify_outputs is skipped.
  RunAndVerifyOutputsWithEP(ort_model_path, "TestNhwcConvReluClipFusion_FP16", std::move(ep), feeds, params, {}, false);
```

Reply:
thx, fixed

Reply:
So far, CPU EP doesn't implement FP16 Clip fusion. The verification will fail because it looks CPU EP will use FP32 Clip.
To verify the Xnnpack FP16 conv fusion correctness, I add a new test with a new FP16 model ( with only Conv+Relu).
Current test is kept because we need to make sure that Conv+Clip fusion can run.

Reply:
XNNPack EP Conv implementation also handles QLinearConv doesn't it?